parlai/core/utils.py

```diff
@@ -713,6 +715,52 @@ def __contains__(self, key):
         """Determine if text contains any offensive words in the filter."""
         return self.contains_offensive_language(key)
 
+    def str_segment(self, text, dict_agent):
+        freqs = dict_agent.freqs()
+
+        # Total number of word tokens
+        N = sum(freqs.values())
+
+        # Number of distinct words in the Vocab
+        V = len(freqs)
+
+        @lru_cache(maxsize=None)
+        def segment(text):
+            # Return a list of words that is the best segmentation of text.
+            if not text:
+                return []
+            candidates = ([first]+segment(rem) for first, rem in splits(text))
+            return max(candidates, key=prob_words)
+
+        def splits(text, L=20):
```

Comment:
What's the rationale behind 20? Seems pretty high but maybe it's good in practice.

Reply:
Yes, why 20? this should be an argument of the larger function